packages/manager/apps/public-cloud/src/components/easter-eggs/easter-eggs.controller.js

```diff
@@ -0,0 +1,98 @@
+import angular from 'angular';
+import Konami from 'konami-code-js';
+
+import {
+  EASTER_EGGS_LOCAL_STORAGE_KEY,
+  EASTER_EGGS_LOCAL_STORAGE_KEY_ACTIVATE,
+  EASTER_EGGS_LOCAL_STORAGE_KEY_DEACTIVATE,
+  EASTER_EGGS_TRACKING_KEY,
+} from './easter-eggs.constants';
+
+import controller from './modal/controller';
+import template from './modal/template.html';
+
+export default class EeasterEggsController {
+  /* @ngInject */
+  constructor($document, $timeout, $uibModal, atInternet) {
+    this.$document = $document;
+    this.$timeout = $timeout;
+    this.$uibModal = $uibModal;
+    this.atInternet = atInternet;
+  }
+
+  $onInit() {
+    this.rootElement = this.$document[0].querySelector('html');
+
+    if (EeasterEggsController.isDarkThemeActive()) {
+      this.rootElement.classList.add('theme-dark');
+    }
+
+    this.konami = new Konami(() => this.toggle());
+  }
+
+  /**
+   * Check if the dark theme is activate by looking into the localStorage value.
+   * @return {Boolean}
+   */
+  static isDarkThemeActive() {
+    return localStorage
```

Comment:
Let’s use me/preferences for this kind of parameters :)

If our customers open  a new window or private tab as example, the dark mode will disappear. I think it’s better to keep it :)

Reply:
We can discuss about it when we will reach the Beta for this feature.

For now I will explicitly set this value into the localStorage as it was already the case for the current language for instance..

Reply:
@jleveugle i think you though about sessionStorage but localStorage act like cookies, they persist even when the browser is closed